client/pool_test.go

```diff
@@ -0,0 +1,76 @@
+package client
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+	"testing"
+)
+
+func TestAddPoolAndDeletePool(t *testing.T) {
+	c := NewClient(GetConfigFromEnv())
+
+	name := "testacc"
+	ranges := "172.16.0.1-172.16.0.8,172.16.0.10"
+	comment := "terraform-acc-test-pool"
+	expectedPool := &Pool{
+		Name:    name,
+		Ranges:  ranges,
+		Comment: comment,
+	}
+	pool, err := c.AddPool(
+		name,
+		ranges,
+		comment,
+	)
+
+	fmt.Sprintf("expected pool:  %v", expectedPool)
+	fmt.Sprintf("created pool:  %v", pool)
+	if err != nil {
+		t.Errorf("Error creating a pool with: %v", err)
+	}
+
+	if len(pool.Id) < 1 {
+		t.Errorf("The created pool does not have an Id: %v", pool)
+	}
+
+	if strings.Compare(pool.Name, expectedPool.Name) != 0 {
```

Comment:
You can do `if pool.Name != expectedPool.Name`. There is probably other code that does the pattern you mentioned but it's not as idiomatic.

Let's do that for all of these string comparisons